test/contentAsset.test.ts

```diff
@@ -0,0 +1,184 @@
+    const tokenId = receipt.logs[0].topics[3];
+    return tokenId;
+  }
+
+  async function deployContentAssetFixture(): Promise<ContentAssetFixture> {
+    await hre.deployments.fixture(['ContentAsset']);
+    const accounts = await hre.ethers.getSigners();
+    const ContentAsset = await hre.ethers.getContract<ContentAsset>('ContentAsset');
+    const ContentAssetStorage = await hre.ethers.getContract<ContentAssetStorage>('ContentAssetStorage');
+    const ServiceAgreementV1 = await hre.ethers.getContract<ServiceAgreementV1>('ServiceAgreementV1');
+    const Token = await hre.ethers.getContract<Token>('Token');
+
+    return { accounts, ContentAsset, ContentAssetStorage, ServiceAgreementV1, Token };
+  }
+
+  beforeEach(async () => {
+    ({ accounts, ContentAsset, ContentAssetStorage, ServiceAgreementV1, Token } = await loadFixture(
+      deployContentAssetFixture,
+    ));
+  });
+
+  it('The contract is named "ContentAsset"', async function () {
+    expect(await ContentAsset.name()).to.equal('ContentAsset');
+  });
+
+  it('The contract is version "1.0.0"', async function () {
+    expect(await ContentAsset.version()).to.equal('1.0.0');
+  });
+
+  it('Create an asset, send 0 assertionId, expect to fail', async () => {
+    assetInputStruct.assertionId = ZERO_BYTES32;
+    await expect(ContentAsset.createAsset(assetInputStruct)).to.be.revertedWith('Assertion ID cannot be empty');
+    assetInputStruct.assertionId = assertionId;
+  });
+
+  it('Create an asset, send size 0, expect to fail', async () => {
+    assetInputStruct.size = 0;
+    await expect(ContentAsset.createAsset(assetInputStruct)).to.be.revertedWith('Size cannot be 0');
+    assetInputStruct.size = 1000;
+  });
+
+  it('Create an asset, send 0 epochs number, expect to fail', async () => {
+    assetInputStruct.epochsNumber = 0;
+    await expect(ContentAsset.createAsset(assetInputStruct)).to.be.revertedWithCustomError(
+      ServiceAgreementV1,
+      'ZeroEpochsNumber',
+    );
+    assetInputStruct.epochsNumber = 5;
+  });
+
+  it('Create an asset, send 0 token amount, expect to fail', async () => {
+    assetInputStruct.tokenAmount = 0;
+    await expect(ContentAsset.createAsset(assetInputStruct)).to.be.revertedWithCustomError(
+      ServiceAgreementV1,
+      'ZeroTokenAmount',
+    );
+    assetInputStruct.tokenAmount = 250;
+  });
+
+  it('Create an asset, expect asset created', async () => {
+    await Token.mint(accounts[0].address, 1000000000000000);
+    await Token.increaseAllowance(ServiceAgreementV1.address, assetInputStruct.tokenAmount);
+
+    await expect(ContentAsset.createAsset(assetInputStruct)).to.emit(ContentAsset, 'AssetMinted');
+  });
+
+  it('Get an non existing asset, expect 0 returned', async () => {
+    expect(await ContentAssetStorage.getAssertionIds(nonExistingTokenId)).to.deep.equal([]);
+  });
+
+  it('Get an existing asset, expect asset returned', async () => {
```

Comment:
Also here, we're getting basically root hashes of the asset states (assertion identifiers), not really asset itself